gateway/handler_success.go

```diff
@@ -332,6 +338,21 @@ func (s *SuccessHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) *http
 	return nil
 }
 
+func stripListenPath(r *http.Request, listenPath string) error {
+	tmp := new(mux.Route).PathPrefix(listenPath)
+	s, err := tmp.GetPathRegexp()
+	if err != nil {
+		return err
+	}
+
+	reg := regexp.MustCompile(s)
```

Comment:
Not sure about this particular scenario but our `regexp` package might be useful here? https://github.com/TykTechnologies/tyk/blob/master/regexp/regexp.go

cc @dencoded

Reply:
It is already used, see above